lib/api/core/plugins/pluginsManager.js

```diff
@@ -419,14 +426,18 @@ class PluginsManager {
    *
    * @param {object} plugin
    * @param {string} event
-   * @param {string} fn - function to attach
+   * @param {string|Function} fn - function to attach
    */
   registerHook(plugin, event, fn) {
     debug('[%s] register hook on event "%s"', plugin.manifest.name, event);
 
     this.kuzzle.on(event, message => {
       try {
-        plugin.object[fn](message, event);
+        if (_.isFunction(fn)) {
+          fn(message, event);
```

Comment:
why not 
```javascript
fn.call(plugin.object, message, event)
```
This would allow to define the piped function as just its reference, like

```javascript
pipes: {
  action1: this.handler1
}
```

without losing the context of the plugin.

Reply:
For my part I think it's better to let the developer to choose which context he want to be binded with his function. Javascript developers knows that they should carefully handle the scope when passing function, an implicit binding may lead to confusion.
I think we should deprecate the old way to declare pipes, hooks and controller's actions and document only the way with the arrow function.

Reply:
I think your point is legit, but I still think this needs further discussion with the team. I'd need a workshop to get more opinions on this.

Reply:
My thoughts: 
* we have to expect many novice JS developers asking us questions since, contrary to SDKs, plugin developers cannot choose their preferred language, and they are stuck with Javascript
* but, apart from security reasons, we should never make expectations about what a plugin developer want or don't want. For instance, a plugin developer could decide to create a separate object for a specific pipe, and forcing the context to the main plugin object would lead to unnexpected (and, frankly, irritating) behaviors

=> we should add a notice in our plugin documentation and/or how-tos and/or boilerplates, but we shouldn't babysit developers